10up/component-library
# Component Library v2

This PR consolidates all of the baseline components in a single repo and does a bunch of improvements:

- Adds unit tests with jest, those tests simulate a real user using the application rather than testing for implementation details.
- Run the axe accessibility checker as part of the jest tests, ensuring the markup is valid after component initialization and during interactions with the component.
- Replace the few puppeteer tests with cypress as it's easier to maintain as has a much better developer experience, tests are recorded in Github and also in the cypress dashboard. The new tests actually test how a real user would interact with the components.
- Only load the polyfills necessary from core-js 3 instead of loading babel-polyfill. Bundle sizes decreased quite a bit. See the table below:


| Component  | Bundle Size | Previous Bundle Size |  Reduction |
| ------------- | ------------- | ------------- | ------------- |
| Accordion  | 3KB  | 16KB  | 81.25% |
| Navigation | 6KB  | 27KB  | 77.7% |
| Countdown Timer| 6KB  | 37KB  | 83.7% | 
| Tabs | 4KB  | 20KB  | 80% |
| Tooltips | 4KB  | 23KB  | 82.6% |

- Add TS definitions, so IDE like VSCode will now get autocompletion for free (even if not using TS)
![component-library-autocomplete](https://user-images.githubusercontent.com/6104632/107850421-17d77f00-6de1-11eb-9997-cda992eb2422.gif)


The components implementation are mostly intact, I didn't introduce new features or made significant changes besides very small fixes.

**Steps for testing**
- Pull down this branch
- npm install (make sure you have npm 7)
- `npm run build`
- `cd e2e/ && npm run start`
- Visit localhost:9090

The original demo folders are still present in each component, so the umd exports can be tested there.

**Next Steps:**

- [x] Move Issues from other repo to this repo
- [ ] Release v2 packages within the `next` tag in npm
- [ ] Release final v2 packages
- [ ] Archive the individual component repos.
